control-plane/connect-inject/endpoints_controller.go

```diff
@@ -57,19 +50,29 @@ const (
 	// exposedPathsStartupPortsRangeStart is the start of the port range that we will use as
 	// the ListenerPort for the Expose configuration of the proxy registration for a startup probe.
 	exposedPathsStartupPortsRangeStart = 20500
+
+	// ConsulNodeName is the node name that we'll use to register and deregister services.
+	ConsulNodeName = "k8s-service-mesh"
+
+	// ConsulNodeAddress is the address of the consul node (defined by ConsulNodeName).
+	// This address does not need to be routable as this node is ephemeral, and we're only providing it because
+	// Consul's API currently requires node address to be provided when registering a node.
+	ConsulNodeAddress = "127.0.0.1"
+
+	// ConsulKubernetesCheckType is the type of health check in Consul for Kubernetes readiness status.
+	ConsulKubernetesCheckType = "kubernetes-readiness"
+
+	// ConsulKubernetesCheckName is the name of health check in Consul for Kubernetes readiness status.
+	ConsulKubernetesCheckName = "Kubernetes Readiness Check"
```

Comment:
in a custom health check like this, is the status of the check all consul would care about to decide whether to route traffic to the app?

Reply:
yes!

Reply:
thanks!!